mm/kasan/kasan.c

```diff
@@ -79,6 +82,11 @@ static FAR uintptr_t *kasan_mem_to_shadow(FAR const void *ptr, size_t size,
   FAR struct kasan_region_s *region;
   uintptr_t addr = (uintptr_t)ptr;
 
+  if (g_region_init != KASAN_INIT_VALUE)
```

Comment:
Why not just use `bool`?

Reply:
The code instrumented by compiler's KASAN may run before the startup initialize .bss or .data section. Using the bigger number increase the possiblity to catch the premature access before KASAN subsystem finish the initialization.

Reply:
So basically `g_region_init` can be a random value. But what will happen if the device is reseted, but memory content retains?

Reply:
it depends on the hardware design, so the approach just adds an additional safeguard.